hbase-client/src/main/java/org/apache/hadoop/hbase/client/OnlineLogRecord.java

```diff
@@ -52,6 +59,15 @@ final public class OnlineLogRecord extends LogEntry {
         if (slowLogPayload.getMultiServiceCalls() == 0) {
           jsonObj.remove("multiServiceCalls");
         }
+        if (slowLogPayload.getScan().isPresent()) {
+          try {
+            jsonObj.add("scan", JsonParser.parseString(slowLogPayload.getScan().get().toJSON()));
```

Comment:
We have to add a JsonElement either way, so I think with toMap we'd still have to convert the map to json here

Reply:
Can't you do this?

```
jsonObj.add("scan", gson.toJsonTree(scan.toMap()));
```

I think this is categorically better than `JsonParser.parseString(slowLogPayload.getScan().get().toJSON())`. The reason is because the current approach requires going `scan -> map -> JsonElement -> string -> JsonElement -> String` (the latter two coming from JsonParser and then the serialization of the resulting `jsonObj`. The approach recommended above just has to go `scan -> map -> JsonElement -> string`.

Reply:
Oh yeah, you're right. Will make this change! Thanks for the tip